src/typecheck.hpp

```diff
@@ -4,12 +4,10 @@ namespace TypedAST {
 struct TypedAST;
 }
 
-namespace Frontend {
-struct CompileTimeEnvironment;
-}
-
 namespace TypeChecker {
 
+struct TypeChecker;
+
```

Comment:
Also a great deal of sense, since we are _typechecking_, to use the _TypeChecker_

Reply:
When you change an abstraction and realize that things suddenly make sense: it's just so... nice.